postgres/configure-and-start-postgres.sh

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env bash
+#
+# Ensures that Postgres is installed on this machine.
+# Creates DB and user.
+# Configures Postgres to accept host connections to new Database.
+
+set -eu
+
+SCRIPT_DIR="$( realpath --strip "$( dirname "${BASH_SOURCE[0]}" )" )"
+source "$(dirname ${SCRIPT_DIR})/common.sh"
+
+set -e
+set -u
+
+############################
+### Arguments processing ###
+############################
+
+usage() {
+    echo "Usage: ${0} --host <HOST> --dbname <DBNAME> --username <USERNAME> \\"
+    echo "            --password <USER_PWD> --data-dir <PATH>"
+    echo
+    echo "Options:"
+    echo "   --host <HOST>          Host IP to accept connections on"
+    echo "   --dbname <DBNAME>      Database name to create"
+    echo "   --username <USERNAME>  Role name to create"
+    echo "   --password <USER_PWD>  Password to use for new user"
+    echo "   --data-dir <PATH>      Postgres data dir path (default: Postgres default)."
+    exit 1
+}
+
+HOST=
+DBNAME=
+USERNAME=
+PASSWORD=
+DATA_DIR=
+
+# Let's get the command line arguments.
+while [ $# -gt 0 ]; do
+    if [ "${1}" = "--host" ]; then
+        shift; if [ -z "${1}" ]; then usage; fi
```

Comment:
This test will fail when `${1}` does not exist (due to shift), i.e. use `${1:-}`